supportedBackends/spark/src/main/scala/cromwell/backend/impl/spark/SparkClusterProcess.scala

```diff
@@ -0,0 +1,173 @@
+package cromwell.backend.impl.spark
+
+import java.nio.file.Path
+
+import akka.actor.ActorSystem
+import cromwell.backend.impl.spark.SparkClusterProcess.{ParserResponse, TerminalStatus}
+import spray.http.{HttpRequest, HttpResponse, StatusCodes}
+import spray.json.{DefaultJsonProtocol, JsonParser}
+import spray.client.pipelining._
+
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import better.files._
+import com.typesafe.scalalogging.Logger
+import org.slf4j.LoggerFactory
+import spray.httpx.unmarshalling._
+
+import scala.util.{Failure, Success, Try}
+
+object SparkClusterProcess {
+  sealed trait SparkClusterResponse
+  case class SuccessResponse(action: String, driverState: String, serverSparkVersion: String,
+                             submissionId: String, success: Boolean, workerHostPort: String, workerId: String) extends SparkClusterResponse
+  sealed trait ParserResponse
+  case class SuccessfulParsedResponse(action: String, message: String, serverSparkVersion: String, submissionId: String, success: Boolean) extends ParserResponse
```

Comment:
Failure response is possible from the parse function but since it is used inside a future container we got that exception and hence onFailure of that future.